### ISSUE TITLE
MSSQL Clustered ColumnStore indices can not be queried

### ISSUE THREAD

**Issue by roa-nyx** [dbeaver/dbeaver#5527]
I have not found any open issue that seems relevant to this.

#### System information: 
Windows 10 (latest updates as of March 18, 2019)
DBeaver 6.0.0.201903101945
Java (no idea)
Additional extensions: mssql driver

#### Connection specification:
Microsoft SQL Server 2017 (RTM-CU5) (KB4092643) - 14.0.3023.8 (X64)   Mar  2 2018 18:24:44   Copyright (C) 2017 Microsoft Corporation  Web Edition (64-bit) on Windows Server 2016 Datacenter 10.0 <X64> (Build 14393: ) (Hypervisor) 
Driver name: Microsoft JDBC Driver for SQL Server
No tunnels or proxies

#### Describe the problem you're observing:
I can't query tables that have a clustered columnstore index.

```
SELECT DISTINCT [FieldA] FROM [ColStoreTable]
/* results in:
Query execution failed

Reason:
SQL Error [35370] [S0001]: Cursors are not supported on a table which has a clustered columnstore index.
*/
```

#### Steps to reproduce, if exist:
Create a table in an MS-SQL server with a clustered columnstore index, and try to query it.

It seems to me that DBeaver uses a cursor to iterate through results, but this isn't allowed for clustered columnstore indexes. Not sure how this can be addressed.

#### Include any warning/errors/backtraces from the logs
I'm skipping the logs here as I feel there is enough information here and I don't want to try to anonymize/sanitize any potential leakages from the log files however if you feel you do end up needing the log files please let me know.

**Comment by wgrusznis** [comment]
I have the same problem.
I tried the latest Microsoft JDBC driver without any success. 
This makes DBeaver unusable with clustered columnstore indexed tables.

as a workaround I use 
SELECT DISTINCT top 10 [FieldA] FROM [ColStoreTable]

I always thought 'Use SQL to limit fetch size' does exactly this but it seems I was wrong.

**Comment by marcushenningsen** [comment]
I can confirm this problem. My workaround is to use SSMS, but I'd much prefer to consolidate my SQL work in DBeaver

**Comment by marcoracer** [comment]
I can confirm this problem. We use Azure SQL Database and when we create a clustered columnstore index and query it we get:

Query execution failed

Reason:
SQL Error [35370] [S0001]: Cursors are not supported on a table which has a clustered columnstore index.

**Comment by higorhi** [comment]
Having the same issue as of today, with Azure SQL Database. 
I'm on Solus OS 4.0, running DBeaver 6.0.5.201905202310.

**Comment by marcushenningsen** [comment]
Don't know why, but I just found out that using "top 10" (or any other number) in the select statement works around the issue.

**Comment by Nietsnut** [comment]
Setting the 'ResultSet fetch size' to 0 in the preferences solves this problem

**Comment by wgrusznis** [comment]
it doesn't solve the problem if you would like to use fetch size different then 0.
And this is the problem.

**Comment by Marex-me** [comment]
The issue still persists
MS-SQL
macOS Catalina 10.15.5 
DBeaver 7.1.1.202006211844

**Comment by serge-rider** [comment]
Try to set "Use SQL to limit fetch size parameter":
![image](https://user-images.githubusercontent.com/6398845/93177057-df048280-f73a-11ea-8cea-5a9ded38a3b5.png)

**Comment by roa-nyx** [comment]
While I do have this level of control over the staging server, I will need to provide good reason to making this modification on the production server, and the fact that I want to use a different tool than everyone else just isn't something I'll be able to defend with my team.

For me at least @serge-rider's workaround is a non-starter.

**Comment by kseniiaguzeeva** [comment]
@roa-nyx , @wgrusznis , @marcoracer , @Marex-me Maybe you have any ideas how to clarify which tables option **Use SQL to limit fetch size** shoud be enabled for?

**Comment by roa-nyx** [comment]
Ok so I tried @serge-rider's suggestion and it worked well.

I don't think I really knew anything about this option when I first posted about this, but now that I know about it I don't really like fetch-size. My solution will be to turn it off permanently.

@kseniiaguzeeva The only way to know that I can think of is when the DBeaver queries for intellisense objects, it could determine which ones have a COLUMNSTORE index, and disable this functionality for those tables auto-magically. Alternatively DBeaver can wait for this error to happen, and then swallow the error and retry the query without fetch-size limits, or using the SQL version of fetch-size limits.

Sorry, I don't think either of these are practical, but it's the best I've got :(
